wandb/sdk/wandb_lite_run.py

```diff
@@ -0,0 +1,344 @@
+                    "summaryMetrics": "{}",
+                    "tags": [],
+                    "description": None,
+                    "notes": None,
+                    "state": "running",
+                },
+            )
+
+            self.i_api.set_current_run_id(self._run.id)
+
+        return self._run
+
+    @property
+    def stream(self) -> file_stream.FileStreamApi:
+        if self._stream is None:
+            # Setup the FileStream
+            self._stream = file_stream.FileStreamApi(
+                self.i_api, self.run.id, datetime.datetime.utcnow().timestamp()
+            )
+            self._stream._client.headers.update(
+                {"X-WANDB-USE-ASYNC-FILESTREAM": "true"}
+            )
+            self._stream.set_file_policy(
+                "wandb-history.jsonl",
+                file_stream.JsonlFilePolicy(start_chunk_id=0),
+            )
+            self._stream.start()
+
+        return self._stream
+
+    @property
+    def pusher(self) -> FilePusher:
+        # TODO: figure out how to trigger file uploads
+        if self._pusher is None:
+            self._pusher = FilePusher(self.i_api, self.stream)
+
+        return self._pusher
+
+    def log_artifact(
+        self,
+        artifact: typing.Union[Artifact, StrPath],
+        name: typing.Optional[str] = None,
+        type: typing.Optional[str] = None,
+        aliases: typing.Optional[typing.List[str]] = None,
+    ) -> Artifact:
+        if aliases is None:
+            aliases = []
+        if not isinstance(artifact, Artifact):
+            raise AttributeError("We don't support logging paths in this interface")
+        artifact.finalize()
+        ## TODO: use a cleaner interace here
+        manifest_dict = _manifest_json_from_proto(
+            InterfaceBase()._make_artifact(artifact).manifest  # type: ignore[abstract]
+        )
+        saver = artifact_saver.ArtifactSaver(
+            api=self.i_api,
+            digest=artifact.digest,
+            manifest_json=manifest_dict,
+            file_pusher=self.pusher,
+            is_user_created=False,
+        )
+        saver.save(
+            type=artifact.type,
+            name=artifact.name,
+            client_id=artifact._client_id,
+            sequence_client_id=artifact._sequence_client_id,
+            metadata=artifact.metadata,
+            description=artifact.description or None,
+            aliases=["latest"] + aliases,
+            use_after_commit=False,
+            finalize=True,
+        )
+        return artifact
+
+    def log(
+        self,
+        row_dict: typing.Dict[str, typing.Any],
+        step: typing.Optional[int] = None,
+        commit: typing.Optional[bool] = None,
+        sync: typing.Optional[bool] = None,
+    ) -> None:
+        stream = self.stream
+        row_dict = {
+            **row_dict,
+            "_timestamp": datetime.datetime.utcnow().timestamp(),
+        }
+        processed_dict = history_dict_to_weave(self, row_dict)
```

Comment:
non-blocking comment: seems like `history_dict_to_weave` should be part of StreamTable, and LiteRun should just call dumps

Reply:
Yeah, I can move it.